server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/SumcoinindexRateSource.java

```diff
@@ -0,0 +1,152 @@
+/*************************************************************************************
+ * Copyright (C) 2014-2016 GENERAL BYTES s.r.o. All rights reserved.
+ *
+ * This software may be distributed and modified under the terms of the GNU
+ * General Public License version 2 (GPL2) as published by the Free Software
+ * Foundation and appearing in the file GPL2.TXT included in the packaging of
+ * this file. Please note that GPL2 Section 2[b] requires that all works based
+ * on this software must also be made publicly available under the terms of
+ * the GPL2 ("Copyleft").
+ *
+ * Contact information
+ * -------------------
+ *
+ * GENERAL BYTES s.r.o.
+ * Web      :  http://www.generalbytes.com
+ *
+ ************************************************************************************/
+package com.generalbytes.batm.server.extensions.extra.sumcoin.sources.sumcoinindex;
+
+import com.generalbytes.batm.server.extensions.Currencies;
+import com.generalbytes.batm.server.extensions.extra.sumcoin.sources.sumcoinindex.*;
+import com.generalbytes.batm.server.extensions.Currencies;
+import com.generalbytes.batm.server.extensions.IRateSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.net.ssl.SSLContext;
+import java.security.NoSuchAlgorithmException;
+import java.security.KeyManagementException;
+import si.mazi.rescu.RestProxyFactory;
+import si.mazi.rescu.ClientConfig;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+public class SumcoinindexRateSource implements IRateSource{
+    private static final Logger log = LoggerFactory.getLogger(SumcoinindexRateSource.class);
+
+    private static HashMap<String,BigDecimal> rateAmounts = new HashMap<String, BigDecimal>();
+    private static HashMap<String,Long> rateTimes = new HashMap<String, Long>();
+    private static final long MAXIMUM_ALLOWED_TIME_OFFSET = 30 * 1000; //30sec
+
+    private String preferedFiatCurrency = Currencies.USD;
+    private ISumcoinindexAPI api;
+
+    public SumcoinindexRateSource(String preferedFiatCurrency) {
+        this();
+        if (Currencies.EUR.equalsIgnoreCase(preferedFiatCurrency)) {
+            this.preferedFiatCurrency = Currencies.EUR;
+        }
+        if (Currencies.USD.equalsIgnoreCase(preferedFiatCurrency)) {
+            this.preferedFiatCurrency = Currencies.USD;
+        }
+    }
+
+    public SumcoinindexRateSource() {
+	final ClientConfig config = new ClientConfig();
+        try {
+            SSLContext sslcontext=SSLContext.getInstance("TLS");
```

Comment:
I think after your previous changes the `SSLContext` and `ClientConfig` is not needed anymore. You should also be able to remove the try-catch leaving the `api` assignment as the only statement in the constructor. This will simplify the constructor considerably.

Reply:
Done